packages/workflow/src/workflow.ts

```diff
@@ -1143,15 +1146,17 @@ export function defineQuery<Ret, Args extends any[] = [], Name extends string =
  *
  * @param def an {@link UpdateDefinition}, {@link SignalDefinition}, or {@link QueryDefinition} as returned by {@link defineUpdate}, {@link defineSignal}, or {@link defineQuery} respectively.
  * @param handler a compatible handler function for the given definition or `undefined` to unset the handler.
+ * @param options an optional `description` of the handler and an optional update `validator` function.
  */
 export function setHandler<Ret, Args extends any[], T extends SignalDefinition<Args> | QueryDefinition<Ret, Args>>(
```

Comment:
Explode that overload into a `setHandler` for signal definition, and a distinct one for query definition.

Reply:
Done